runtime/kphp_tracing.h

```diff
@@ -0,0 +1,332 @@
+  void addAttributeEnum(const string &key, int64_t enum_id, int64_t value) const;
+
+  class_instance<C$KphpSpanEvent> addEvent(const string &name, const Optional<double> &manual_timestamp) const;
+  void addLink(const class_instance<C$KphpSpan> &another) const;
+
+  void updateName(const string &title, const string &short_desc) const;
+  void finish(const Optional<double> &manual_timestamp) const;
+  void finishWithError(int64_t error_code, const string &error_msg, const Optional<double> &manual_timestamp) const;
+  void exclude() const;
+};
+
+inline void f$KphpSpan$$addAttributeString(const class_instance<C$KphpSpan> &v$this, const string &key, const string &value) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->addAttributeString(key, value);
+  }
+}
+inline void f$KphpSpan$$addAttributeInt(const class_instance<C$KphpSpan> &v$this, const string &key, int64_t value) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->addAttributeInt(key, value);
+  }
+}
+inline void f$KphpSpan$$addAttributeFloat(const class_instance<C$KphpSpan> &v$this, const string &key, double value) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->addAttributeFloat(key, value);
+  }
+}
+inline void f$KphpSpan$$addAttributeBool(const class_instance<C$KphpSpan> &v$this, const string &key, bool value) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->addAttributeBool(key, value);
+  }
+}
+inline void f$KphpSpan$$addAttributeEnum(const class_instance<C$KphpSpan> &v$this, const string &key, int64_t enum_id, int64_t value) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->addAttributeEnum(key, enum_id, value);
+  }
+}
+inline class_instance<C$KphpSpanEvent> f$KphpSpan$$addEvent(const class_instance<C$KphpSpan> &v$this, const string &name, const Optional<double> &manual_timestamp = {}) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    return v$this->addEvent(name, manual_timestamp);
+  }
+  return {};
+}
+inline void f$KphpSpan$$addLink(const class_instance<C$KphpSpan> &v$this, const class_instance<C$KphpSpan> &another) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->addLink(another);
+  }
+}
+inline void f$KphpSpan$$updateName(const class_instance<C$KphpSpan> &v$this, const string &title, const string &short_desc) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->updateName(title, short_desc);
+  }
+}
+inline void f$KphpSpan$$finish(const class_instance<C$KphpSpan> &v$this, const Optional<double> &manual_timestamp = {}) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->finish(manual_timestamp);
+  }
+}
+inline void f$KphpSpan$$finishWithError(const class_instance<C$KphpSpan> &v$this, int64_t error_code, const string &error_msg, const Optional<float> &manual_timestamp = {}) {
+  if (kphp_tracing::cur_trace_level >= 1) {
+    v$this->finishWithError(error_code, error_msg, manual_timestamp);
+  }
+}
+inline void f$KphpSpan$$exclude(const class_instance<C$KphpSpan> &v$this) {
```

Comment:
For example, when we've started a span, and then decided, that we don't want to see it in the output. (it will be preserved in binlog of course, but won't be rendered in UI)

Example 1:
```
// top of the function
$span = KphpTracing::startSpan(...);
...
if ($invalid_input) {
  $span->exclude(); 
  return;
}
...
$span->finish();
```

Example 2:
```
/**
 * @kphp-tracing
 */
function renderFeed() {
  if ($GLOBALLY_DISABLED) {
    KphpTracing::getCurrentActiveSpan()->exclude();
    return;
  }
}
```